stock_warehouse_flow/models/stock_warehouse_flow.py

```diff
@@ -0,0 +1,367 @@
+                "active": True,
+                "flow_id": self.id,
+                # Give the priority to routes/rules not tied to a flow
+                # when the initial move is created by the procurement
+                "sequence": 1000,
+            }
+        )
+        self.delivery_route_id = self.env["stock.location.route"].create(vals)
+        self.to_picking_type_id = self.out_type_id
+        # Subscribe the route to the warehouse
+        if self.delivery_route_id.warehouse_selectable:
+            self.warehouse_id.route_ids |= self.delivery_route_id
+
+    def action_generate_route(self):
+        for flow in self:
+            flow._generate_delivery_route()
+        return True
+
+    def _search_for_move_domain(self, move):
+        return [
+            ("from_picking_type_id", "=", move.picking_type_id.id),
+            ("carrier_ids", "in", move.group_id.carrier_id.ids),
+            ("delivery_route_id", "!=", False),
+        ]
+
+    def _is_valid_for_move(self, move):
+        self.ensure_one()
+        if not self.move_domain:
+            return self
+        domain = safe_eval(self.move_domain or "[]")
+        if not domain:
+            return self
+        return move.filtered_domain(domain)
+
+    @api.model
+    def _search_for_move(self, move):
+        """Return the first flow matching the move."""
+        domain = self._search_for_move_domain(move)
+        for flow in self.search(domain):
+            if flow._is_valid_for_move(move):
+                return flow
+        return self.browse()
+
+    def _get_rule_from_delivery_route(self, html_exc=False):
+        rule = self.delivery_route_id.rule_ids.filtered(
+            lambda r: r.picking_type_id == self.to_picking_type_id
+        )
+        if self.delivery_route_id and not rule:
+            args = (
+                self.to_picking_type_id.display_name,
+                self.delivery_route_id.display_name,
+            )
+            if html_exc:
+                args = (
+                    f"<strong>{self.to_picking_type_id.display_name}</strong>",
+                    f"<strong>{self.delivery_route_id.display_name}</strong>",
+                )
+            raise UserError(
+                _(
+                    "No rule corresponding to '%s' operation type "
+                    "has been found on delivery route '%s'.\n"
+                    "Please check your configuration."
+                )
+                % args
+            )
+        return rule
+
+    def _apply_on_move(self, move):
+        """Apply the flow configuration on the move."""
+        if not self:
+            return False
+        logger.info("Applying flow '%s' on '%s'", self.name, move)
+        rule = self._get_rule_from_delivery_route()
+        move.picking_id = False
+        move.picking_type_id = self.to_picking_type_id
+        move.location_id = self.to_picking_type_id.default_location_src_id
```

Comment:
As we pull the chain from the output location and that is it now on the flow itself, if we can put a sub-location, we could achieve even more advanced routing with this little change
```suggestion
        move.location_id = self.output_stock_loc_id
```
The location output_stock_loc_id should then never be readonly

Reply:
This needs to be tackled. Changing the output location is the key concept of pulling the new chain of moves (not changing the out picking type). I could keep the warehouse out picking type and only change the output location to alter the pick/pack flow